### DISCUSSION TITLE
SchemaAutoConfiguration not work ?

### DISCUSSION THREAD

**Discussion by lemuellew** [ExpediaGroup/graphql-kotlin#1690] [Q&A] [answered]
I start from https://medium.com/expedia-group-tech/creating-a-reactive-graphql-server-with-spring-boot-and-kotlin-54aca7316470

It seems that `SchemaAutoConfiguration` not work ?
![image](https://user-images.githubusercontent.com/59448437/221797291-01d89ec2-4080-4caf-b369-5a43c6818958.png)

**Comment by dariuszkuc** [comment]
Hello 👋 

Looks like your application started successfully and you should be able to query it using GraphiQL IDE (defaults to `/graphiql`) or directly through `/graphql` endpoint. We recently disabled automatic printing of the schema on startup. You can re-enable it by specifying `graphql.printSchema=true` property (see https://opensource.expediagroup.com/graphql-kotlin/docs/server/spring-server/spring-properties)

Thanks,
Derek

**Comment by samuelAndalon** [comment]
+1 to what @dariuszkuc mentioned, if your problem still persists, we will need more info, not just an screenshot, repro steps, etc --  that way we could troubleshoot your issue.

**Comment by lemuellew** [comment, marked as answer]
`build.gradle.kts`
``` kotlin
import org.jetbrains.kotlin.gradle.tasks.KotlinCompile

plugins {
    id("org.springframework.boot") version "3.0.3"
    id("io.spring.dependency-management") version "1.1.0"
    kotlin("jvm") version "1.7.22"
    kotlin("plugin.spring") version "1.7.22"
}

group = "com.example"
version = "0.0.1-SNAPSHOT"
java.sourceCompatibility = JavaVersion.VERSION_17

repositories {
    mavenCentral()
}

dependencies {
    implementation("org.springframework.boot:spring-boot-starter")
    implementation("org.jetbrains.kotlin:kotlin-reflect")
    implementation("org.jetbrains.kotlin:kotlin-stdlib-jdk8")
    implementation("com.expediagroup", "graphql-kotlin-spring-server", "6.4.0")
    testImplementation("org.springframework.boot:spring-boot-starter-test")
}

tasks.withType<KotlinCompile> {
    kotlinOptions {
        freeCompilerArgs = listOf("-Xjsr305=strict")
        jvmTarget = "17"
    }
}

tasks.withType<Test> {
    useJUnitPlatform()
}
```
---

`HelloWorldQuery`
``` kotlin
package com.example.graphql

import com.expediagroup.graphql.server.operations.Query
import org.springframework.stereotype.Component

@Component
class HelloWorldQuery: Query {
    fun helloWorld() = "Hello World!"
}
```
---

```
  .   ____          _            __ _ _
 /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
 \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
  '  |____| .__|_| |_|_| |_\__, | / / / /
 =========|_|==============|___/=/_/_/_/
 :: Spring Boot ::                (v3.0.3)

2023-03-01T10:57:01.427+08:00  INFO 1780 --- [           main] c.example.graphql.GraphqlApplicationKt   : Starting GraphqlApplicationKt using Java 17.0.4.1 with PID 1780 (C:\Users\xxx\Downloads\graphql\graphql\build\classes\kotlin\main started by xxx in C:\Users\xxx\Downloads\graphql\graphql)
2023-03-01T10:57:01.439+08:00  INFO 1780 --- [           main] c.example.graphql.GraphqlApplicationKt   : No active profile set, falling back to 1 default profile: "default"
2023-03-01T10:57:04.201+08:00  INFO 1780 --- [           main] o.s.b.web.embedded.netty.NettyWebServer  : Netty started on port 8080
2023-03-01T10:57:04.216+08:00  INFO 1780 --- [           main] c.example.graphql.GraphqlApplicationKt   : Started GraphqlApplicationKt in 3.796 seconds (process running for 5.094)
```

![image](https://user-images.githubusercontent.com/59448437/222034322-959fbe62-00b3-45e5-9401-ef0b6d53945b.png)

**Reply by samuelAndalon** [reply]
https://github.com/ExpediaGroup/graphql-kotlin/issues/1679#issuecomment-1437941955

**Reply by samuelAndalon** [reply]
TLDR; 
if you want to use spring-boot 3.0.3 you have to use 7.x.x releases (still on alpha)

**Reply by lemuellew** [reply]
Thanks for the reply, by the way when will the 7.x.x version be released ?

**Reply by samuelAndalon** [reply]
alpha releases are already available -> https://github.com/ExpediaGroup/graphql-kotlin/releases
